### PR TITLE
avoid conflicts with metis.h

### LINKED ISSUE #24: scotch recipe includes metis.h, conflicting with metis itself
scotch installs a metis.h, which I think is meant as a fallback mechanism.

I propose moving this to `include/scotch/metis.h` or just removing it, so it's not conflicting with the 'real' metis package.

### PR DESCRIPTION
**Pull request by minrk**
by putting them in `include/scotch/` so they don’t conflict with the ‘real’ headers from metis/parmetis packages.

closes #24 (for scotch)

### CODE REVIEW

**Comment by minrk** [comment]
@conda-forge-admin please rerender

**Comment by minrk** [comment]
This preserves metis.h in a scotch subdirectory. The alternative is to remove both metis.h and libscotchmetis. I'm not prepared to say that removal is the right thing to do, so I'm leaving it, just fixing the conflict. If someone with more knowledge can say whether removal is the right thing to do, that would be helpful.

**Comment by basnijholt** [comment]
I wouldn't know the consequences of removing or putting it in a different directory to be honest, so I can't say much here.

Maybe @jbweston?

**Comment by jbweston** [comment]
TL;DR

We might as well keep the headers for completeness, but they should in any case be the same as the ones shipped with `metis`.

---

> I wouldn't know the consequences of removing or putting it in a different directory to be honest

As far as I can tell all (from the [user manuals](https://gforge.inria.fr/docman/?group_id=248&view=listfile&dirid=326) `metis` and `parmetis` related stuff in scotch is providing a `metis`-compatible interface to scotch. That is, if someone is using `metis` in their program, they should be able to link against `libscotchmetis` (instead of `libmetis`) and it will "just werk".

I would imagine that the scotch people include the `metis` header files for "completeness", as a public way to announce what parts of the `metis` API they are implementing. If `metis` changes its API, and there is not a corresponding change in `libscotchmetis`, then you'd run into problems using the `metis.h` shipped with `metis` itself if you tried to link against `libscotchmetis`.

Please don't make me say `metis` anymore.

**Comment by jbweston** [comment]
Removal would mean that people who are currently using `metis` would not be able to "drop in" scotch as a replacement. I think this is an insanely niche case, so it might be worth just removing it, and if anyone complains we can just add it back?

**Comment by dalcinl** [comment]
@minrk Given that you already did the hard work, please merge your PRs. The removal is not strictly necessary, and I have no objections to your "better safe than sorry" rationale. After merging your PRs, other packages can depend both on `metis` and `scotch` without conflicts.

**Comment by minrk** [comment]
ok, let's go with this for now, and we'll see if any issues crop up. Removing it and moving it have equivalent breakage in that anything depending on metis.h will not be able to build without an update. However, that update is easy with this move: add `-I$PREFIX/include/scotch`, whereas it's less so for the removal.

I'm not sure if anyone would use libscotchmetis from conda anyway, since I don't know if there's a reason to use it if 'real' metis is available.
